integration-tests/utils/integration-utils.js

```diff
@@ -50,6 +50,11 @@ const clickable = ({ text, textContains, isDescendant = false }) => {
   return getClickablePath(base, text, textContains, isDescendant)
 }
 
+const enabledClickable = ({ text, textContains, isDescendant = false }) => {
+  const base = `(//a[@aria-disabled="false"] | //button[@aria-disabled="false"] | //*[@role="button" and @aria-disabled="false"] | //*[@role="link" and @aria-disabled="false"] | //*[@role="combobox" and @aria-disabled="false"] | //*[@role="option" and @aria-disabled="false"])`
```

Comment:
I don't think that a native button will have "aria-disabled" on it. However, we never create a native button. I did see that we set "aria-disabled" on anchor tags.

I'm not sure what the correct xpath would be for a disabled native button, but I figure that it's not worth figuring out at this time. If we do start using native buttons, and if someone tries to call this method on one, this method will always fail and they can figure out the correct xpath at that time.

Reply:
I think we could roll this into `clickable`. I can't imagine any test case that, when asking for something clickable, would be okay with it being disabled.

Reply:
Possibly, though I definitely would like to do that in a follow-up PR because I really want to merge this one ASAP. Personally, the way I think of this is "I want to wait for this button to become enabled, and then I want to click on it". I would expect the "wait for the button to become enabled" to be an opt-in because the test-writer should know that in the scenario being tested, it is expected that the button will be disabled initially. This is not normally the case. So I'm still trying to work through in my head if it is OK to make this the default behavior of `clickable`.

Reply:
I understand the urgency to merge this PR and I'm okay with that.

After that, I'm very confident that we can and should roll this into `clickable`. The original intent was for the tests to read simply and clearly, so the extra cognitive overhead of having to be aware of `clickable` and `enabledClickable` feels cumbersome. Looking through the code, of all the calls to `clickable` there are 6 cases that don't match `click (page, clickable(...))`. 3 of those are just inserting a `waitForNoSpinners` between finding and clicking. 2 (consecutive) ares looking for status updates to the cloud environment widget and are presumably using `clickable` to avoid finding the words "Creating" or "Running" elsewhere on the page. The last one is coming from FireCloud and needs to hack a navigation back to the Terra UI under test (localhost, PR test deploy, etc) instead of necessarily going to the shared Terra UI (dev, alpha, etc).

Reply:
@breilly2 I'm coming around to it… I will see about rolling it into my next "QA-1700 PR" that is forthcoming.

Reply:
Personally, I am ok with merging this as-is to help alleviate immediate developer pain and improve the testing signal. But taking another look - I believe this xpath will do the equivalent:

```suggestion
const base = '(//a | //button | //*[@role="button"] | //*[@role="link"] | //*[@role="combobox"] | //*[@role="option"])[@aria-disabled="false"]'
```

My first reaction was the same as @breilly2. But I was thinking about the use cases that @cahrens pointed out and didn't comment. Would re-naming help, along with an explicit wait? Perhaps something like: 

```javascript
await clickableEnabled(page, {text: 'foo'})
await click(page, clickable({text: 'foo'})
```

When combined with the xpath change, this could just be an option to `clickable` - perhaps `isDisabled`

I do like the explicit nature of checking the clickable as this is something a user would do and it makes the test read nicely. I can follow along in the app this way and see that the test will wait for the appropriate state before proceeding - it looks very deliberate to me even if a little verbose. I could see a case where we may want to ensure something is not clickable until the inputs are verified.

Reply:
I personally like having "clickableEnabled" as a separate, explicit call. I took a crack at what I think @petesantos is suggesting in https://github.com/DataBiosphere/terra-ui/pull/2811/commits/948a201efcbe4be09b940239255bae2db8e74d7f. @breilly2 what do you think?

Reply:
FWIW, I really don't think that one would ever have "aria-disabled" on a native button because buttons have disabled as a property built-in. So including "aria-disabled" by default on an xpath that is supposed to handle a button feels wrong to me, though in reality I don't think we ever use native buttons so it is probably a non-issue (😞 ).